testing/cloudbuild-templates/cloud-run-template.cloudbuild.yaml

```diff
@@ -0,0 +1,70 @@
+steps:
+
+- id: 'Build Container Image'
+  name: 'gcr.io/cloud-builders/docker:latest'
+  args: ['build', '-t', 'gcr.io/${PROJECT_ID}/${_SERVICE}:${SHORT_SHA}', '.']
+  dir: '${_SAMPLE_DIR}'
+
+- id: 'Push Container Image'
+  name: 'gcr.io/cloud-builders/docker:latest'
+  args: ['push', 'gcr.io/${PROJECT_ID}/${_SERVICE}:${SHORT_SHA}']
+
+- id: 'Deploy to Cloud Run'
+  name: 'gcr.io/cloud-builders/gcloud:$_CLOUDSDK_VERSION'
+  args:
+  - 'run'
+  - 'deploy'
+  - '${_SERVICE}-${SHORT_SHA}'
```

Comment:
Given the need to keep the overall length of the service name in check, it makes sense to use `SHORT_SHA` rather than full `COMMIT_SHA`. However, there are some cases to consider in how SHORT_SHA might be reused:

1. Re-running the pipeline to address an intermittent failure. This seems like it would happen, but would the repeat be a problem?
2. A totally separate pipeline doing a test that involves creating its own container image or deploying the service. We can work around this in the off chance that time comes.
3. Opening a second PR from the same branch targeting a different destination branch. This is not a workflow I anticipate us ever needing on this repo, so it is a low risk problem.

In the public issue [b/156023473](https://buganizer.corp.google.com/issues/156023473) I've suggested maybe a unix timestamp would work. Simon is also considering generating a random string.

I think we should still use the SHORT_SHA with the container image tag so we can follow code provenance.

Reply:
If the Cloud Run name length gets sorted out, we could use [$BUILD_ID](https://cloud.google.com/cloud-build/docs/configuring-builds/substitute-variable-values) as differentiator.

Reply:
Is this the last item that is preventing this PR from being merged?

Reply:
Yes. We need to resolve this, maybe with timestamp or random generated characters.

Reply:
updated to generate random characters